arch/riscv/include/asm/mach-dpu/pci.h

```diff
@@ -138,56 +137,144 @@
+#define PCIE_CORE_X4_0_CFG0_START     (PCIE_CORE_X4_1_ADDR_START - PCIE_CORE_CFG_SIZE)
+#define PCIE_CORE_X4_0_CFG1_START     (PCIE_CORE_X4_1_ADDR_START - 2*PCIE_CORE_CFG_SIZE)
+#define PCIE_CORE_X4_1_CFG0_START     (PCIE_SUBSYS_ADDR_END - PCIE_CORE_CFG_SIZE)
+#define PCIE_CORE_X4_1_CFG1_START     (PCIE_SUBSYS_ADDR_END - 2*PCIE_CORE_CFG_SIZE)
 
 #ifdef CONFIG_DPU_PCIE_TEST
 #define MSI_INT     0
 #define INTA_INT    1
 #endif
 
+#define PCIE_RC_ADDR_LOW_REG 0x824
+#define PCIE_RC_ADDR_HI_REG 0x828
+#define PCIE_EP_ADDR_LOW_REG 0x82c
+#define PCIE_EP_ADDR_HI_REG 0x834
+#define PCIE_HUGE_DATA_SIZE_REG 0x838
+#define PCIE_RC_RCVED_REG 0x840
+#define PCIE_EP_RCVED_REG 0x844
+#define PCIE_EP_WR_SIZE_PERTIME_REG 0x84c
+
+#define PCIE_DDR_BASE_ADDR_LOW_REG 0x820
+#define PCIE_DDR_BASE_ADDR_HI_REG 0x850
+
+#define PCIE_CMD_REG 0x858
+/*
+ * reg[2:0]:
+ * 1: mem alloc
+ * 2: mem free
+ * 3: mem dump
+ * 7: invalid cmd
+ */
+#define MEM_ALLOC_REQ 1
+#define MEM_FREE_REQ 2
+#define MEM_DUMP_REQ 3
+#define INVALID_CMD_REQ 7
+
+#define PCIE_TOTAL_SIZE_LW_REG 0x85c
+/*
+ * reg[31:0]:total_size[31:0]
+ */
+
+#define PCIE_TOTAL_SIZE_HI_REG 0x864
+/*
+ * reg[3:0]:total_size[35:32]
+ */
+
+#define PCIE_FLAG_REG 0x868
+/*
+ * reg[2:0]
+ * for rc2ep:
+ *      0: last package finished
+ *      1: malloc requrest
+ *      4: run mcu_code
+ * for ep2rc:
+ *      2:ep return malloc response
+ *      3:ep last package finished
+ *      5:ep return free response
+ *      6:ep run mcu code ok response
+ */
+#define EP2RC_INVALID_FLG 0
+#define RC2EP_LAST_PKG_FLG 1
+#define RC2EP_MEM_ALLOC_REQ_FLG 2
+#define EP2RC_MALLOC_OK_FLG 3
+#define EP2RC_LAST_PKG_FLG 4
+#define RC2EP_RUN_MCU_CODE_FLG 5
+#define EP2RC_FREE_OK_FLG 6
+#define EP2RC_RUN_MCU_OK_FLG 7
+
+#define EP_ALLOCED_MEM_LW_REG 0x870
+/*
+ * reg[31:0]: addr[31:0]
+ */
+
+#define EP_ALLOCED_MEM_HI_REG 0x874
+/*
+ * reg[3:0]: addr[35:32]
+ */
+#define RC_FREE_MEM_LW_REG 0x87c
+/*
+ * reg[31:0]: addr[31:0]
+ */
+
+#define RC_FREE_MEM_HI_REG 0x880
+/*
+ * reg[3:0]: addr[35:32]
+ * reg[7:4]: reserved page for invalid DMA
+ */
+
+#define EP_RSVED_MEM_LW_REG 0x888
+/*
+ * reg[31:0]: addr[31:0]
+ */
+
 #ifdef IPBENCH
 void apb_read_c(uint32_t addr, uint32_t *data, int port);
 void apb_write_c(uint32_t addr, uint32_t data, int port);
 #endif
 
 void init_duowen_pcie_subsystem(void);
 
-struct duowen_pcie_subsystem
-{
-    uint64_t cfg_apb[5];
-    uint8_t link_mode;
+struct duowen_pcie_subsystem {
+	uint64_t cfg_apb[5];
+	uint8_t link_mode;
 
 #ifdef IPBENCH
-    //uint8_t port_
+	//uint8_t port_
 #endif
 
-    struct dw_pcie *controller;
+	struct dw_pcie *controller;
 
-    //uint8_t (* get_link_mode)(struct duowen_pcie_subsystem *pcie_subsystem);
-    //uint32_t (* apb_read)(uint64_t *addr);
-    //void (* apb_write)(uint64_t *addr, uint32_t val);
+	//uint8_t (* get_link_mode)(struct duowen_pcie_subsystem *pcie_subsystem);
+	//uint32_t (* apb_read)(uint64_t *addr);
+	//void (* apb_write)(uint64_t *addr, uint32_t val);
 };
```

Comment:
OK.
But since indentation changes messes the patch, please split all indentation changes into 1 single patch.